src/sonic-config-engine/portconfig.py

```diff
@@ -114,9 +114,15 @@ def get_port_config(hwsku=None, platform=None, port_config_file=None):
         if not port_config_file:
             return ({}, {})
 
+
     # Read from 'platform.json' file
     if port_config_file.endswith('.json'):
-        hwsku_json_file = get_hwsku_file_name(hwsku, platform)
+        if not hwsku_config_file:
+             hwsku_json_file = get_hwsku_file_name(hwsku, platform)
+             if not hwsku_config_file:
```

Comment:
First, this should be "if not hwsku_json_file" ?

The behavior is changed here.  If we don't specify hwsku.json and it does not exist,  should we return empty or raise?  Not sure how many caller use this function, can they handle empty return? I think we should  raise.

Reply:
Modified the line to "if not hwsku_json_file". 
If we don't specify hwsku.json and it does not exist, we should return empty as it's the expected behavior of **"get_port_config"** from beginning. Whoever calls  "get_port_config" function, should handle an empty return. That's the current behavior of code. 

Let me know if I am missing something.